src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1202,6 +1205,100 @@ private async Task<WebResponse> SendRequest(bool async)
+                            break;
+                        case HttpRequestCacheLevel.Default:
+                            if (httpRequestCachePolicy.MinFresh > TimeSpan.Zero)
+                            {
+                                request.Headers.CacheControl.MinFresh = httpRequestCachePolicy.MinFresh;
+                            }
+
+                            if (httpRequestCachePolicy.MaxAge != TimeSpan.MaxValue)
+                            {
+                                request.Headers.CacheControl.MaxAge = httpRequestCachePolicy.MaxAge;
+                            }
+
+                            if (httpRequestCachePolicy.MaxStale > TimeSpan.Zero)
+                            {
+                                request.Headers.CacheControl.MaxStale = true;
+                                request.Headers.CacheControl.MaxStaleLimit = httpRequestCachePolicy.MaxStale;
+                            }
+
+                            break;
+                        case HttpRequestCacheLevel.Refresh:
+                            request.Headers.CacheControl.MaxAge = TimeSpan.Zero;
+                            request.Headers.Pragma.Add(new NameValueHeaderValue("no-cache"));
+                            break;
+                    }
+                }
+                else
+                {
+                    switch (policy.Level)
+                    {
+                        case RequestCacheLevel.NoCacheNoStore:
+                            request.Headers.CacheControl.NoCache = true;
+                            request.Headers.CacheControl.NoStore = true;
+                            request.Headers.Pragma.Add(new NameValueHeaderValue("no-cache"));
+                            break;
+                        case RequestCacheLevel.Reload:
+                            request.Headers.CacheControl.NoCache = true;
+                            request.Headers.Pragma.Add(new NameValueHeaderValue("no-cache"));
+                            break;
+                        case RequestCacheLevel.CacheOnly:
+                            request.Headers.CacheControl.OnlyIfCached = true;
+                            break;
+                    }
+                }
+            }
+        }
+
+        private RequestCachePolicy? GetApplicableCachePolicy()
+        {
+            if (CachePolicy != null)
+            {
+                return CachePolicy;
+            }
+            else if (IsDefaultCachePolicySet(DefaultCachePolicy))
+            {
+                return DefaultCachePolicy;
+            }
+            else if (IsDefaultCachePolicySet(WebRequest.DefaultCachePolicy))
+            {
+                return WebRequest.DefaultCachePolicy;
+            }
+
+            return null;
+        }
+
+        private static bool IsDefaultCachePolicySet(RequestCachePolicy? policy) => policy != null
+                        && policy.Level != RequestCacheLevel.BypassCache;
```

Comment:
Actually, can you make sure `BypassCache` doesn't need to say anything?

The docs say it bypasses all caches "between client and server", which imply if there is a proxy set, we need to treat it as `Refresh`. From a cursory glance, it's not clear if Framework is actually doing this to me or if docs are wrong or treating "server" to include proxies.

Reply:
> This doesn't seem correct: if I explicitly want my request to bypass cache, this will end up using the default policy (which might be to cache).
> 
> I think correct behavior is to stop at the first non-null policy, and return null if that policy says bypass.

I did it because the DefaultCachePolicy is initialized with BypassCache in WebRequest and HttpWebRequest by default. if a user sets WebRequest.DefaultCachePolicy and didn't do it for HttpWebRequest.DefaultCachePolicy we will presume that no cache gonna be applied, which may seem misleading

Reply:
> I did it because the DefaultCachePolicy is initialized with BypassCache in WebRequest and HttpWebRequest classes by default. if a user sets WebRequest.DefaultCachePolicy and didn't do it for HttpWebRequest.DefaultCachePolicy we will presume that no cache gonna be applied, which may seem misleading for the user

It sounds like we need to update those properties to track "explicitly set" vs "default" state. Pull the first one that is explicitly set.